common/util.go

```diff
@@ -265,3 +269,73 @@ func (s *ServiceClient) DatalakeServiceClient() (*datalake.Client, error) {
 	}
 	return s.dsc, nil
 }
+
+// This is currently used only in testcases
+func NewServiceClient(bsc *blobservice.Client,
+					  fsc *fileservice.Client,
+					  dsc *datalake.Client) *ServiceClient {
+	return &ServiceClient {
+		bsc: bsc,
+		fsc: fsc,
+		dsc: dsc,
+	}
+}
+
+type FileClientStub interface {
+	URL() string
+}
+
+// DoWithOverrideReadOnly performs the given action, and forces it to happen even if the target is read only.
+// NOTE that all SMB attributes (and other headers?) on the target will be lost,
+// so only use this if you don't need them any more
+// (e.g. you are about to delete the resource, or you are going to reset the attributes/headers)
+func DoWithOverrideReadOnlyOnAzureFiles(ctx context.Context, action func() (interface{}, error), targetFileOrDir FileClientStub, enableForcing bool) error {
```

Comment:
Can we add some UT for this method?

Reply:
The function has been unchanged. That should be taken as a diffrent review, does not fit to the purpose of this one

Reply:
We need to add a new test.  Testing is required to prevent regressions.

Reply:
Narasimha is right, it is old functionality, but it should still have a test.

Reply:
Posted a separate review here https://github.com/Azure/azure-storage-azcopy/pull/2540